snap/info_snap_yaml_test.go

```diff
@@ -39,14 +41,801 @@ type: app
+
+func (s *YamlSuite) TestUnmarshalStandaloneImplicitPlug(c *C) {
+	// NOTE: yaml content cannot use tabs, indent the section with spaces.
+	info, err := snap.InfoFromSnapYaml([]byte(`
+name: snap
+plugs:
+    network-client:
+`))
+	c.Assert(err, IsNil)
+	c.Check(info.Name, Equals, "snap")
+	c.Check(info.Plugs, HasLen, 1)
+	c.Check(info.Slots, HasLen, 0)
+	plug := info.Plugs["network-client"]
+
+	c.Assert(plug, Not(IsNil))
+	c.Check(plug.Snap, Equals, info)
+	c.Check(plug.Name, Equals, "network-client")
+	c.Check(plug.Interface, Equals, "network-client")
+	c.Check(plug.Attrs, HasLen, 0)
+	c.Check(plug.Label, Equals, "")
+	c.Check(plug.Apps, HasLen, 0)
+}
+
+func (s *YamlSuite) TestUnmarshalStandaloneAbbreviatedPlug(c *C) {
+	// NOTE: yaml content cannot use tabs, indent the section with spaces.
+	info, err := snap.InfoFromSnapYaml([]byte(`
+name: snap
+plugs:
+    net: network-client
+`))
+	c.Assert(err, IsNil)
+	c.Check(info.Name, Equals, "snap")
+	c.Check(info.Plugs, HasLen, 1)
+	c.Check(info.Slots, HasLen, 0)
+	plug := info.Plugs["net"]
+
+	c.Assert(plug, Not(IsNil))
+	c.Check(plug.Snap, Equals, info)
+	c.Check(plug.Name, Equals, "net")
+	c.Check(plug.Interface, Equals, "network-client")
+	c.Check(plug.Attrs, HasLen, 0)
+	c.Check(plug.Label, Equals, "")
+	c.Check(plug.Apps, HasLen, 0)
+}
+
+func (s *YamlSuite) TestUnmarshalStandaloneMinimalisticPlug(c *C) {
+	// NOTE: yaml content cannot use tabs, indent the section with spaces.
+	info, err := snap.InfoFromSnapYaml([]byte(`
+name: snap
+plugs:
+    net:
+        interface: network-client
+`))
+	c.Assert(err, IsNil)
+	c.Check(info.Name, Equals, "snap")
+	c.Check(info.Plugs, HasLen, 1)
+	c.Check(info.Slots, HasLen, 0)
+	plug := info.Plugs["net"]
+
+	c.Assert(plug, Not(IsNil))
+	c.Check(plug.Snap, Equals, info)
+	c.Check(plug.Name, Equals, "net")
+	c.Check(plug.Interface, Equals, "network-client")
+	c.Check(plug.Attrs, HasLen, 0)
+	c.Check(plug.Label, Equals, "")
+	c.Check(plug.Apps, HasLen, 0)
+}
+
+func (s *YamlSuite) TestUnmarshalStandaloneCompletePlug(c *C) {
+	// NOTE: yaml content cannot use tabs, indent the section with spaces.
+	info, err := snap.InfoFromSnapYaml([]byte(`
+name: snap
+plugs:
+    net:
+        interface: network-client
+        ipv6-aware: true
+`))
+	c.Assert(err, IsNil)
+	c.Check(info.Name, Equals, "snap")
+	c.Check(info.Plugs, HasLen, 1)
+	c.Check(info.Slots, HasLen, 0)
+	plug := info.Plugs["net"]
+
+	c.Assert(plug, Not(IsNil))
+	c.Check(plug.Snap, Equals, info)
+	c.Check(plug.Name, Equals, "net")
+	c.Check(plug.Interface, Equals, "network-client")
+	c.Check(plug.Attrs, DeepEquals, map[string]interface{}{"ipv6-aware": true})
+	c.Check(plug.Label, Equals, "")
+	c.Check(plug.Apps, HasLen, 0)
+}
+
+func (s *YamlSuite) TestUnmarshalLastPlugDefinitionWins(c *C) {
+	// NOTE: yaml content cannot use tabs, indent the section with spaces.
+	info, err := snap.InfoFromSnapYaml([]byte(`
+name: snap
+plugs:
+    net:
+        interface: network-client
+        attr: 1
+    net:
```

Comment:
Yeah, originally this test was supposed to fail but well... parsers are parsers.